common/app/model/TopMentionsModel.scala

```diff
@@ -20,7 +20,23 @@ object TopMentionsResponse {
   implicit val TopMentionsDetailsJf: Format[TopMentionsDetails] = Json.format[TopMentionsDetails]
 }
 
-case class TopMentionsTopic(`type`: TopMentionsTopicType, value: String)
+trait Topic {
+  def `type`: TopMentionsTopicType
+  def value: String
+}
+
+case class TopicWithCount(
+    `type`: TopMentionsTopicType,
+    value: String,
+    count: Int,
+) extends Topic
```

Comment:
I'm not sure what would be the best way, but I think instead of creating the trait you could also just extend the base model like this:
```
case class TopicWithCount(
    override val `type`: TopMentionsTopicType,
    override val value: String,
    count: Int,
) extends TopMentionsTopic(`type`, value)
```

@alinaboghiu what do you think would be the recommended way of doing this? Or maybe both ways are equally OK?